Update Hardware docs (#174)
srobo/runbook: docs/kit/hardware/kch.md

# KCH (Raspberry Pi HAT)

![3D Render of KCH](img/kch-3d-render.png)

## KCH LEDs

The LEDs on the KCH are split into several groups, some indicate the power state of the KCH, some indicate the state of the robot and some are directly controlled by student code.

### Power Supply

- `5V` (Green)
    - 5V power is present, whether from the USB-C port on the Pi or the KCH regulator.
- `Reg` (Green)
    - The regulator on the KCH is enabled.
- `Power in` (Green)
    - 12V power is present on the 12V input.
- `Reverse polarity` (Red)

>     - The 12V power input has reversed polarity.

Reviewer: ```suggestion
    - The 12V power input has reversed polarity. The 12V input wires should be swapped over to correctly power the Pi.
```

### Boot Progress LEDs

There are 5 green LEDs that are a "progress bar" of boot progress on the KCH. Each represents a different stage of boot:

- 20%
    - Connected to GPIO 7, which is configured as [SPI0 CE1 (ALT0)](https://pinout.xyz/pinout/pin26_gpio7)
    - It is configured in the [KCH EEPROM](https://github.com/srobo/kch-hw/blob/main/EEPROM/eeprom_settings.txt#L69).

>     - In normal behaviour, it turns on when the EEPROM device tree overlay is loaded, i.e in the early stages of the kernel.

Reviewer: Boot 20% should light even with an invalid kernel, as long as start*.elf is present.
```suggestion
    - In normal behaviour, it turns on when the GPIO map is loaded from the EEPROM, i.e in the videocore bootloader (which requires a start*.elf).
```

Author: Thanks Will. I had wrongly assumed that the device tree was only applied when a kernel was actually present to execute.

Reviewer: This change doesn't look to have been applied.

Author: > This change doesn't look to have been applied.

I did it manually rather than using the GitHub UI as I had other manual changes to make too.

    - This LED indicates that the SD card can be read and that the operating system is loading.
    - This LED will turn on with any valid Raspberry Pi distro.
- 40%
    - This LED is turned on by [`boot_40.service`](https://github.com/srobo/robot-image/blob/main/sources/meta-srobo/recipes-robot/python-astoria/astoria-config/boot_40.service), which runs in the early stages of systemd (`basic.target`).

>     - This LED indicates that the filesystem on the SD card is valid.

Author: Filesystem here refers to the actual data on the card, not the FAT partition. i.e there is a valid Linux kernel, device tree and systemd is able to run. 

An invalid FAT partition would be indicated by the ACT LED on the Pi flashing. See above table.

>     - This LED indicates that the filesystem on the SD card is valid.

Author: 13d167c

    - It also indicates that the SD card contains a Student Robotics OS image.
- 60%
    - This LED is controlled by [`kchd`](../../software/#kch-daemon-kchd) and indicates `kchd` is running.
    - If this LED has turned on, most of the operating system is loaded and behaving correctly.
- 80%

>     - This LED is turned on when the MQTT Event Broker is running

Author: Neither. It turns on when `kchd` connects to the MQTT broker.

>     - This LED is turned on when the MQTT Event Broker is running

Author: However, I'm not sure that level of detail is useful for this page.

>     - This LED is turned on when the MQTT Event Broker is running

Author: 13d167c

- 100%
    - This LED is turned on when all critical astoria services are loaded and running.
    - It will turn off if any of the services stop or crash.

### Status LEDs

The following LEDs are driven by GPIO on the Raspberry Pi and indicate the status of the robot.

> All except the `Start` and Heartbeat LED are controlled by [`kchd`](../../software/#kch-daemon-kchd). If `kchd` is not running, these LEDs will not light up. Check the boot progress LEDs to confirm.

Author: 13d167c

- `Code` (Green)

>     - Enabled when a USB drive containing a usercode file (`robot.py`) is mounted.

Author: It will still light up. It is on when any drive containing a `robot.py` is mounted and known by Astoria.

> something critical like a syntax error

This is an error that occurs in the Python process, so would be indicated by the `OK` LED turning red. This is because a syntax error results in a negative exit code.

- `Comp` (Blue)
    - Enabled when the `mode` key in the robot metadata is `COMP`.
    - This does not indicate that the arena marker offset is enabled, or that the WiFi is disabled.
- `WiFi` (Blue)
    - Enabled when the WiFi Access Point is running.

> - Heartbeat (Blue)

Reviewer 2: style nit: "Hearbeat" here is not backtick quoted -- is that intentional?

Author: Yes. The heartbeat is symbolised by a heart _symbol_ on the board rather than literal words as the other indicators are.

Reviewer 2: Perhaps
```suggestion
- Heartbeat `❤` (Blue)
```
